PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.azuretools.hdinsight.projects;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.SubProgressMonitor;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.internal.ui.wizards.NewWizardMessages;
+import org.eclipse.jdt.internal.ui.wizards.buildpaths.BuildPathsBlock;
+import org.eclipse.jdt.ui.wizards.JavaCapabilityConfigurationPage;
+import org.eclipse.jdt.ui.wizards.NewJavaProjectWizardPageOne;
+import org.eclipse.jdt.ui.wizards.NewJavaProjectWizardPageTwo;
+import org.scalaide.core.SdtConstants;
+
+public class HDInsightScalaPageTwo extends NewJavaProjectWizardPageTwo {
+	private HDInsightsScalaProjectWizard parent = null;
+	public HDInsightScalaPageTwo(HDInsightScalaPageOne hdInsightScalaPageOne) {
+		super(hdInsightScalaPageOne);
+	}
+	
+	public void setParent(HDInsightsScalaProjectWizard parent) {
+		this.parent = parent;
+	}
+
+	public void configureJavaProject(String newProjectCompliance, IProgressMonitor monitor)
+			throws CoreException, InterruptedException {
+		if (monitor == null) {
+			monitor = new NullProgressMonitor();
+		}
+		byte nSteps = 6;
+		((IProgressMonitor) monitor).beginTask(NewWizardMessages.JavaCapabilityConfigurationPage_op_desc_java,
+				nSteps);
+		try {
+			IProject project = addHDInsightNature(monitor);
+			Method method = JavaCapabilityConfigurationPage.class.getDeclaredMethod("getBuildPathsBlock");
```

Comment:
Actually it is just moved from the wizard to here.

---
In reply to: [132864164](https://github.com/Microsoft/azure-tools-for-java/pull/884#discussion_r132864164) [](ancestors = 132864164)